habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -258,6 +260,181 @@ def calculate_walk_pose(
         self.obj_transform_base = obj_transform_base @ rot_offset
         self.joint_pose = joint_pose
 
+    def calculate_walk_pose_directional(
+        self,
+        target_position: mn.Vector3,
+        distance_multiplier=1.0,
+        target_dir=None,
+    ):
```

Comment:
This new `calculate_walk_pose_directional` is the `calculate_walk_pose` function that was used in P0.
It ensures that the human avatar follows the transforms coming from the VR headset (which always slightly moves around, and has a forward gaze constraint).

I created a new function to avoid changing how the humanoids move during training or in the other applications.

@xavierpuigf FYI - I'll let you decide what to do with this code moving forward.